### PR TITLE
Ruby group capstone - Catalog of my things

### PR DESCRIPTION
**Pull request by shafiuyushawu**
### In this pull request the following tasks were implemented :

- Each class from the [UML class diagram](https://github.com/Jolak5/Catalog-Ruby/pull/images/catalog_of_my_things.png) should be created in a separate .rb file.
- All associations (1-to-many relationships) and aggregations (parent-child relationships) between classes should be implemented.
- All properties visible in the diagram should be defined and set up in the constructor method. Exception: properties for the 1-to-many relationships should NOT be set in the constructor method. Instead, they should have a custom setter method created.
- All methods visible in the diagram should be implemented:
   - add_item method (in all classes that have that method)
      - should take an instance of the Item class as an input.
      - should add the input item to the collection of items.
      - should add self as a property of the item object (by using the correct setter from the item object).
   - can_be_archived?() in the Item class
      - should return true if published_date is older than 10 years.
      - otherwise, it should return false.
   - move_to_archive() in the Item class
       - should reuse can_be_archived?() method.
       - should change the archived property to true if the result of the can_be_archived?() method is true.
       - should do nothing if the result of the can_be_archived?() method is false.
   - can_be_archived?() in the Book class
      - should override the method from the parent class.
      - should return true if parent's method returns true OR if cover_state equals to "bad".
      - otherwise, it should return false.
   - can_be_archived?() in the MusicAlbum class
      - should override the method from the parent class.
      - should return true if parent's method returns true AND if on_spotify equals true.
      - otherwise, it should return false.
   - can_be_archived?() in the Game class
      - should override the method from the parent class.
      - should return true if parent's method returns true AND if last_played_at is older than 2 years.
      - otherwise, it should return false.
- Add unit tests for all implemented methods.

**User interface**
- Create a main.rb file that will serve as your console app entry-point.
- Your console app, at the start, should:
   - Present the user with a list of options to perform.
   - Let users choose an option.
   - If needed, ask for parameters for the option.
   - Have a way to quit the app.
- The following options should be available:
   - List all books
   - List all music albums
   - List of games
   - List all genres (e.g 'Comedy', 'Horror')
   - List all labels (e.g. 'Gift', 'New')
   - List all authors (e.g. 'Stephen King')
   - Add a book
   - Add a music album
   - Add a game
- All data should be preserved by saving collections in .json files.

**Database**

- Create a schema.sql file with tables that will be analogical to the structure of the classes in your app. As you cannot implement inheritance in the database tables 

And [here](https://drive.google.com/file/d/1tLMGq8C9POkrx3c6JqiyN5Tx4UMuDswa/view?usp=sharing) is the link of our presentation.

### CODE REVIEW

File: classes/item.rb [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  def move_to_archive?
+    @archived = true if can_be_archived?
+    @archived
+  end
+
+  def author=(author)
+    @author = author
+    author.items.push(self) unless author.items.include?(self)
+  end
+
```
Comment (emmanuellekamwa) [inline classes/item.rb:23]: - [ ] The item class should include the `Genre`, `Author`, and `Label` methods respectively. Please add them to your item class just as you added the author method.

**Review by emmanuellekamwa — CHANGES_REQUESTED** [review]
**STATUS: CHANGES REQUESTED ⛔❌🛑**
Hello @shafiuyushawu, @edubew and @Matthieu96Code 👋
Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- Good commit messages.
- Descriptive PR.

## Required Changes ♻️
_Check the comments under the review._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: classes/item.rb (hunk 1 above)
Comment (shafiuyushawu) [reply classes/item.rb:23]: Hello @emmanuellekamwa 
Appreciate the feedback, we are working on it.
Thank you

Comment (shafiuyushawu) [reply classes/item.rb:23]: Resolved

**Comment by enniopo** [comment]
Hi @shafiuyushawu, @edubew, @Matthieu96Code Your dashboard has been updated and you can now request another code review. Please go ahead and request a new code review request. 
Apologies for any inconvenience caused.

**Review by Munsa1 — COMMENTED** [review]
Hi @shafiuyushawu @edubew @Matthieu96Code, 
congratulations-2

In my opinion, your project is ready for the presentation!  There is nothing else to say other than... it's time to merge it :shipit: 


Good luck! 🎉 


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**Comment by shafiuyushawu** [comment]
Hello @Munsa1 

Thanks for the approval.

**Pull request merged**